src/diffusers/pipelines/pixart_alpha/pipeline_pixart_alpha.py

```diff
@@ -495,6 +534,28 @@ def prepare_latents(self, batch_size, num_channels_latents, height, width, dtype
         latents = latents * self.scheduler.init_noise_sigma
         return latents
 
+    @staticmethod
+    def classify_height_width_bin(height: int, width: int, ratios: dict) -> Tuple[int, int]:
+        """Returns binned height and width."""
+        ar = float(height / width)
+        closest_ratio = min(ratios.keys(), key=lambda ratio: abs(float(ratio) - ar))
+        default_hw = ratios[closest_ratio]
+        return int(default_hw[0]), int(default_hw[1])
+
+    @staticmethod
+    def resize_and_crop_tensor(samples: torch.Tensor, new_width: int, new_height: int) -> torch.Tensor:
+        orig_hw = torch.tensor([samples.shape[2], samples.shape[3]])
+        custom_hw = torch.tensor([new_height, new_width])
+
+        if (orig_hw != custom_hw).all():
+            ratio = max(custom_hw[0] / orig_hw[0], custom_hw[1] / orig_hw[1])
+            resized_width = int(orig_hw[1] * ratio)
+            resized_height = int(orig_hw[0] * ratio)
+            transform = T.Compose([T.Resize((resized_height, resized_width)), T.CenterCrop(custom_hw.tolist())])
+            return transform(samples)
```

Comment:
This is not worth a `torchvision` dependency IMO. `torchvision` is very heavy and we can do this operations without torchvision